core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -871,6 +872,7 @@ private[spark] class TaskSchedulerImpl(
       tid: Long,
       taskResult: DirectTaskResult[_]): Unit = synchronized {
     taskSetManager.handleSuccessfulTask(tid, taskResult)
+    cleanupTaskState(tid)
```

Comment:
the `handleSuccessfulTask` callback is executed in `task-result-getter` thread pool, will this cause concurrency issues?

Reply:
Yes, all its updates seem to be wrapped with `synchronized`